### PR TITLE
feat: added caller role test suite

### PR DESCRIPTION
**Pull request by Fatimanaz**
Caller Role Access Tests for following contracts https://github.com/Citadel-DAO/citadel-contracts/issues/155
Funding
CitadelMinting
KnightingRound
SupplySchedule
Staking

### CODE REVIEW

File: src/test/GlobalAccessControl.t.sol [hunk 1]
```diff
@@ -334,4 +334,158 @@ contract GlobalAccessControlTest is BaseFixture {
...
+        knightingRound.sweep(address(wbtc));
+
+        vm.warp(block.timestamp + knightingRound.saleDuration());
+        vm.expectRevert("GAC: invalid-caller-role");
+        knightingRound.finalize();
+        vm.prank(governance);
+        knightingRound.finalize();
+    }
+
+    function testStakingCallerRoles() public {
```
Comment (sajanrajdev) [inline src/test/GlobalAccessControl.t.sol]: misleading test name, maybe `testVestingCallerRoles` is more appropiate

File: src/test/GlobalAccessControl.t.sol [hunk 2]
```diff
@@ -334,4 +334,158 @@ contract GlobalAccessControlTest is BaseFixture {
...
+    function testFundingCallerRoles() public {
+        // calling with wrong address
+        vm.expectRevert(bytes("GAC: invalid-caller-role"));
+        fundingCvx.setDiscountLimits(0, 20);
+        vm.prank(address(governance));
+        fundingCvx.setDiscountLimits(10, 50);
+
+        // calling from wrong account
+        vm.expectRevert(bytes("GAC: invalid-caller-role-or-address"));
+        fundingCvx.setDiscount(20);
```
Comment (sajanrajdev) [inline src/test/GlobalAccessControl.t.sol:347]: Can also be called by the `funding.discountManager`

**Comment by sajanrajdev** [comment]
@Fatimanaz Even though access on the xCTDL is not controlled globally it it is still limited locally. I think we should include a test for the access roles on the StakesCitadel.sol contract.

**Comment by Fatimanaz** [comment]
Yeah added them now.

**Review by dapp-whisperer — APPROVED** [review]

**Pull request merged**